test/e2e/elasticsearch_token_propagation_test.go

```diff
@@ -0,0 +1,397 @@
+			Kind:       "Jaeger",
+			APIVersion: "jaegertracing.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "token-prop",
+			Namespace: namespace,
+		},
+		Spec: v1.JaegerSpec{
+			Strategy: "production",
+			Storage: v1.JaegerStorageSpec{
+				Type: "elasticsearch",
+				Elasticsearch: v1.ElasticsearchSpec{
+					NodeCount: 1,
+					Resources: &corev1.ResourceRequirements{
+						Limits:   corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("1Gi")},
+						Requests: corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("1Gi")},
+					},
+				},
+			},
+			Query: v1.JaegerQuerySpec{
+				Options: v1.NewOptions(map[string]interface{}{
+					"es.version":                     "5",
+					"query.bearer-token-propagation": "true",
+					"es.tls":                         "false",
+				}),
+			},
+			Ingress: v1.JaegerIngressSpec{
+				Options: v1.NewOptions(map[string]interface{}{
+					"pass-access-token":      "true",
+					"pass-user-bearer-token": "true",
+					"scope":                  "user:info user:check-access user:list-projects",
+					"pass-basic-auth":        "false",
+				}),
+			},
+		},
+	}
+	return exampleJaeger
+}
+
+func (suite *TokenPropagationTestSuite) deployJaegerWithPropagationEnabled() {
+	queryName := fmt.Sprintf("%s-query", name)
+	collectorName := fmt.Sprintf("%s-collector", name)
+
+	// create jaeger custom resource
+	suite.exampleJaeger = getESJaegerInstance()
+	err := fw.Client.Create(goctx.TODO(),
+		suite.exampleJaeger,
+		&framework.CleanupOptions{
+			TestContext:   ctx,
+			Timeout:       timeout,
+			RetryInterval: retryInterval,
+		})
+	require.NoError(t, err, "Error deploying example Jaeger")
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, collectorName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for collector deployment")
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, queryName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for query deployment")
+
+	route := findRoute(t, fw, name)
+
+	suite.host = route.Spec.Host
+	suite.queryServiceEndPoint = fmt.Sprintf("https://%s/api/traces?service=%s", suite.host, testServiceName)
+
+}
+
+func TestTokenPropagationSuite(t *testing.T) {
+	suite.Run(t, new(TokenPropagationTestSuite))
+}
+
+func newHTTPSClient() *http.Client {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	client := &http.Client{
+		Transport: tr,
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+	return client
+}
+
+func oAuthAuthorization(host, user, pass string) (*http.Client, error) {
+	/* Setup client*/
+	cookieJar, _ := cookiejar.New(nil)
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+		Jar: cookieJar,
+	}
+	/* Start oauth */
+	resp, err := client.Get("https://" + host + "/oauth/start")
```

Comment:
Kinda of the same as previous comments, but you probably just want to retry in case of some well-known networking problems, like timeouts.